pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -165,6 +165,9 @@ let
 
 , ... } @ attrs:
 
+assert attrs ? outputHash ->
+  import ./hash-constraints.nix { inherit lib; } attrs;
+
```

Comment:
There's already an eval-time assertion on line 220, I'd inline the entire check there like this:

```nix
if args.outputHashAlgo or "" == "md5" || lib.hasPrefix "md5-" (outputHash or "") then
  throw "Rejecting insecure ${algo} hash '${outputHash}'"
else
```

This should use less memory.